### DISCUSSION TITLE
Quit replay?

### DISCUSSION THREAD

**Discussion by jungle-boogie** [saulpw/visidata#1350] [Q&A] [answered]
Hello,

I have all the steps in a replay working correctly, except quitting vd. 

I have tried adding `q`, `gq` to the test.vd file I'm using but that still doesn't quit.

I've tried adding a `--replay-wait` delay of 2 seconds, but that also doesn't quit `vd`.

The last thing before I want to quit is saving a file as a csv, which happens. But `vd` does not quit.

**Comment by saulpw** [comment, marked as answer]
Try using `-b` for batch mode.  Then it will automatically quit at the end.  If you specify `-o output.csv` it will save the last sheet to that filename.

**Reply by jungle-boogie** [reply]
Yes! This worked. `vd -b -p test.vd`

Just one caveat: https://github.com/saulpw/visidata/issues/1349

Can vd sessions have variables in them? The file I'm opening contains the date, so ideally I'd like to have the date be a variable.

**Reply by saulpw** [reply]
Yes, use `{varname}` in the .vd or .vdj file, and then specify `varname=value` on the cli when replaying.

**Reply by jungle-boogie** [reply]
That is awesome! Thank you!

`$ vd -b -p test.vd todaydate=2022-04-08`
